sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs

```diff
@@ -62,14 +83,94 @@ internal sealed class CbsTokenProvider : ICbsTokenProvider
         /// <param name="namespaceAddress">The address of the namespace to be authorized.</param>
         /// <param name="appliesTo">The resource to which the token should apply.</param>
         /// <param name="requiredClaims">The set of claims that are required for authorization.</param>
+        ///
         /// <returns>The token to use for authorization.</returns>
         ///
         public async Task<CbsToken> GetTokenAsync(Uri namespaceAddress,
                                                   string appliesTo,
-                                                  string[] requiredClaims)
+                                                  string[] requiredClaims) =>
+            Credential switch
+            {
+                _ when Credential.IsSharedAccessCredential => await AcquireSharedAccessTokenAsync().ConfigureAwait(false),
```

Comment:
We also cannot be sure that a shared key was not rolled/updated within the credential, which would would invalidate our cached token ahead of its expiration, so we don't want to cache these.  The shared access credentials also manage the token lifetime and extend when appropriate, so the only benefit of caching would be to avoid a small number of allocations in a background task doing network IO.